src/cp.js

```diff
@@ -84,18 +84,12 @@ function copyFileSync(srcFile, destFile, options) {
 //
 // Licensed under the MIT License
 // http://www.opensource.org/licenses/mit-license.php
-function cpdirSyncRecursive(sourceDir, destDir, opts) {
+function cpdirSyncRecursive(sourceDir, destDir, currentDepth, opts) {
   if (!opts) opts = {};
 
-  /* Ensure there is not a run away recursive copy. */
-  if (typeof opts.depth === 'undefined') {
-    opts.depth = 0;
-  }
-  if (opts.depth >= common.config.maxdepth) {
-    // Max depth has been reached, end copy.
-    return;
-  }
-  opts.depth++;
+  // Ensure there is not a run away recursive copy
+  if (currentDepth >= common.config.maxdepth) return;
+  currentDepth++;
```

Comment:
Well, ok. I was thinking that `depth` would be the last function parameter, so that you wouldn't have to explicitly call `cpdirSyncRecursive` with it. To initialize the depth:
```
if (!depth) depth = 0;
```
just like we initialize `opts`. Then you can `depth++` or call `cpdirSyncRecursive(..., depth + 1)`, whichever you think is clearer.